applications/StructuralMechanicsApplication/custom_response_functions/adjoint_processes/replace_elements_and_conditions_for_adjoint_problem_process.cpp

```diff
@@ -102,6 +104,18 @@ namespace Kratos
 
                     (*it.base()) = p_element;
                 }
+                else if (element_name == "AdjointFiniteDifferenceTrussElement")
+                {
+                    Element::Pointer p_element = Kratos::make_shared<AdjointFiniteDifferenceTrussElement>(*it.base() );
+
+                    (*it.base()) = p_element;
+                }
+                else if (element_name == "AdjointFiniteDifferenceTrussLinearElement")
+                {
+                    Element::Pointer p_element = Kratos::make_shared<AdjointFiniteDifferenceTrussElementLinear>(*it.base() );
```

Comment:
just for your information:
this calls the assignment constructor if I am not mistaken
`Element::Pointer p_element(Kratos::make_shared<AdjointFiniteDifferenceTrussElementLinear>(*it.base() ));` should call the constructor which I think is more efficient
@msandre is this true?

Reply:
`Element::Pointer p_element = ...` is copy/move constructor with possible implicit type conversion. I had to make a double take when I saw this since it appeared you were trying to start a new shared ptr from an existing one, but I think this is just the way the adjoint element constructs from the primal one. The code will be slightly more resilient to code ripples and implicit conversions if you replace `Element::Pointer` with `auto`.

Reply:
But isn't the conversion from 'AdjointFiniteDifferenceTrussElementLinear::Pointer' to Element::Pointer necessary here?